pkg/library/container/conversion.go

```diff
@@ -108,6 +111,17 @@ func devfileResourcesToContainerResources(devfileContainer *dw.ContainerComponen
 	cpuLimit := devfileContainer.CpuLimit
 	if cpuLimit == "" {
 		cpuLimit = constants.SidecarDefaultCpuLimit
+
+		// Set empty CPU limits when possible:
+		// 1. If there is no LimitRange in the namespace
+		// 2. CPU limits is not overridden
+		// See details at https://github.com/eclipse/che/issues/22198
+		limitRanges := &corev1.LimitRangeList{}
+		if err := k8sClient.List(context.TODO(), limitRanges, &client.ListOptions{Namespace: namespace}); err != nil {
+			return nil, err
+		} else if len(limitRanges.Items) == 0 {
+			cpuLimit = ""
+		}
```

Comment:
Slightly cleaner (set cpuLimit var fewer times)
```suggestion

		// Set empty CPU limits when possible:
		// 1. If there is no LimitRange in the namespace
		// 2. CPU limits is not overridden
		// See details at https://github.com/eclipse/che/issues/22198
		limitRanges := &corev1.LimitRangeList{}
		if err := k8sClient.List(context.TODO(), limitRanges, &client.ListOptions{Namespace: namespace}); err != nil {
			return nil, err
		} else if len(limitRanges.Items) == 0 {
			cpuLimit = ""
		} else {
			cpuLimit = constants.SidecarDefaultCpuLimit
		}
```

Reply:
No idea why the suggestion diffs are so broken today :thinking: